src/main/scala/zio/kafka/admin/AdminClient.scala

```diff
@@ -445,6 +453,27 @@ object AdminClient extends Accessible[AdminClient] {
       }
     }.map(_.asScala.toMap.bimap(TopicPartition(_), ListOffsetsResultInfo(_)))
 
+    /**
+     * List offset for the specified partitions.
+     */
+    override def listOffsetsAsync(
+      topicPartitionOffsets: Map[TopicPartition, OffsetSpec],
+      options: Option[ListOffsetsOptions]
+    ): Task[Map[TopicPartition, Task[ListOffsetsResultInfo]]] = {
+      val asJava                = topicPartitionOffsets.bimap(_.asJava, _.asJava).asJava
+      val topicPartitionsAsJava = topicPartitionOffsets.keySet.map(_.asJava)
+      blocking.effectBlocking {
+        val listOffsetsResult = options
+          .fold(adminClient.listOffsets(asJava))(opts => adminClient.listOffsets(asJava, opts.asJava))
+        topicPartitionsAsJava.map(tp => tp -> listOffsetsResult.partitionResult(tp))
+      }
+    }.map(_.view.map { case (topicPartition, listOffsetResultInfoFuture) =>
```

Comment:
🟢 Since you are doing only one transformation and you force a strict map with `toMap` (so it will be evaluated anyway), I think the `view` is useless (but I might be wrong)
```suggestion
    }.map(_.map { case (topicPartition, listOffsetResultInfoFuture) =>
```

Reply:
`topicPartitionsAsJava.map(tp => tp -> listOffsetsResult.partitionResult(tp))` is a Set.
The view avoids the creation of an intermediary Set before building the final Map.